vita3k/gui/src/controllers_dialog.cpp

```diff
@@ -214,10 +242,17 @@ void draw_controllers_dialog(GuiState &gui, EmuEnvState &emuenv) {
             ImGui::Spacing();
             ImGui::TableSetColumnIndex(2);
             ImGui::TextColored(GUI_COLOR_TEXT_TITLE, "%s", "Motion Support");
+            const char *port_names[] = { "1", "2", "3", "4" };
+            static int selected_port = -1;
             for (auto i = 0; i < ctrl.controllers_num; i++) {
                 ImGui::TableNextRow();
                 ImGui::TableSetColumnIndex(0);
-                ImGui::Text("%-8d", i);
+                selected_port = i;
+                ImGui::PushID(i);
+                ImGui::SetNextItemWidth(50.f * emuenv.dpi_scale);
+                if (ImGui::Combo("##swap_port", &selected_port, port_names, ctrl.controllers_num))
+                    swap_controller_ports(ctrl, i + 1, selected_port + 1);
```

Comment:
```suggestion
                    swap_controller_ports(ctrl, i, selected_port);
```
In `swap_controller_ports`, at 6 different times, you need to decrement the port, while you need the 1-indexed version only twice.

Reply:
hmm i see yes